File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<vector<int>> students = {{1,2,3,4,5}, {2,1,2,3,2,4,2,5}, {3,3,1,1,2,2,4,4,5,5}};
    vector<int> Ocount(3);
    for (int i=0; i < answers.size(); i++)
    {
        for (int j=0 ; j < 3; j++)
            if (answers[i] == students[j][i % students[j].size()]) 
                Ocount[j] += 1;
    }
    
    int maxCount = *max_element(Ocount.begin(),Ocount.end());
    
    for (int i=0 ; i < 3; i++)
        if (Ocount[i] == maxCount) answer.push_back(i+1);    
    
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] answers) {
        List<int> answer = new List<int>();
        int[] array1 = {1, 2, 3, 4, 5};
        int[] array2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] array3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] correctCount = new int[3];
        
        for (int i=0; i<answers.Length; i++)
        {
            if (array1[i % array1.Length] == answers[i]) correctCount[0]++;
            if (array2[i % array2.Length] == answers[i]) correctCount[1]++;
            if (array3[i % array3.Length] == answers[i]) correctCount[2]++;
        }
        
        int max = correctCount.Max();
        
        for (int i=0; i<correctCount.Length; i++)
        {
            if (max == correctCount[i])
                answer.Add(i+1);
        }
        
        return answer.ToArray();
    }
}